### DISCUSSION TITLE
Single Sign On - SSO

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#811] [Q&A]
We have users logging in via a different 3rd party authentication system, but we have other functionality that is user specific that will require the Hasura database. How can we execute a signin to nhost, nhost.auth.signIn, after they have successfully logged in to the 3rd party auth. In other words, users come to a login page, they login with the 3rd party auth, and then they are also logged in to nhost, so that they have a token to interact with hasura. We can't use the admin secret, as these are client side interactions, plus we need to enforce permissions. thanks.

**Comment by elitan** [comment]
That sounds a bit tricky. 

Is it possible to use Nhost's auth system completely? 

If not, what kind of 3rd party auth system is it?

**Reply by osseonews** [reply]
We were looking to integrate this with Shopify's Hydrogen framework (believe this framework will dominate eventually). After a user logs into Shopify, we want to log them into Nhost (need to log in b/c of permissions on the Hasura database). Thanks.

**Reply by elitan** [reply]
I can't think of any easy solution to this. You can check the Auth API definition here: https://github.com/nhost/hasura-auth/blob/main/docs/openapi.json

And render the file using e.g. Swagger Editor here: https://editor.swagger.io/

**Reply by osseonews** [reply]
Couldn't we programatically execute a passwordless signin? How exactly is passwordless generated now. My thinking is that if someone logs in via another system, you just then programatically run a passwordless sigin.  Virtually all Saas now offer SSO, as its really difficult to migrate users in real-life situations (going thru this hell now myself on another app). So in practice, its easier to keep your current login with a provider, and then just SSO into a new provider to get the benefits of user login. I think many people will be in this situation with Nhost. If someone already is using Auth0 to log in customers, it's risky to move (again in theory it's easy, but in reality it's not for various reasons), but you need to SSO into Nhost to use permissions properly on Hasura. Generally, the way SSO works is that after sign in with the original provider, you just sign a user object with a secret key and pass that to the new provider (in this case Nhost). The new provider, Nhost, then verifies the object/key on its end, and if it passes, just signs in the user into the system. 

Of course, in theory the above is easy to do with Hasura, as we can look up the customer in the database after login from the first system and create our own JWT cookie to use for Hasura (we do this now on a few apps), but then we lose the benefits of the Nhost library, which is a huge value here.
